### PR TITLE
feat: warn if nextauth url and port doesn't match

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #<issue>

## ✅ Checklist

- [ ] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [ ] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I performed a functional test on my final commit

---

## Changelog

_[Short description of what has changed]_

---

## Screenshots

_[Screenshots]_

💯

### CODE REVIEW

**Comment by juliusmarminge** [comment]
Current dependencies on/for this PR:
* next
  * **PR #1591** <a href="https://app.graphite.dev/github/pr/t3-oss/create-t3-app/1591" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/t3-oss/create-t3-app/1591?utm_source=stack-comment).

File: cli/template/extras/src/env/with-auth-db.mjs [hunk 1]
```diff
@@ -17,14 +17,27 @@ export const env = createEnv({
...
+    PORT: z.coerce.number().default(3000),
     NEXTAUTH_SECRET:
       process.env.NODE_ENV === "production"
         ? z.string()
         : z.string().optional(),
     NEXTAUTH_URL: z.preprocess(
       // This makes Vercel deployments not fail if you don't set NEXTAUTH_URL
       // Since NextAuth.js automatically uses the VERCEL_URL if present.
-      (str) => process.env.VERCEL_URL ?? str,
+      (str) => {
```
[the remark this answers is not in the document]
Comment (c-ehrlich) [reply cli/template/extras/src/env/with-auth-db.mjs:28]: maintenance burden of adding new files aside, next config feels better to me - here the validation has to run on every request, which seems unnecessary. in next config we could just crash (or log an error, but crashing is harder to ignore) on server start and be done with it.

[the remark this answers is not in the document]
Comment (juliusmarminge) [reply cli/template/extras/src/env/with-auth-db.mjs:28]: nah i think next config is the better choice - put it here purely out of convenience of not having multiple next configs but that's not the best reason 😅

**Pull request closed without merging**